core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -469,4 +472,10 @@ static void setPendingException(ChannelHandlerContext ctx, Throwable cause) {
             logger.warn("{} Unexpected suppressed exception:", ctx.channel(), cause);
         }
     }
+
+    @Override
+    public void initiateConnectionShutdown() {
+        assert responseDecoder != null;
+        responseDecoder.disconnectWhenFinished();
```

Comment:
`responseDecoder.disconnectWhenFinished()` has been removed since the role is similar to `HttpSession.isAcquireable()`. It seems to okay to remove `HttpSession. initiateConnectionShutdown()` and call `HttpSession.deactivate()` instead.

Reply:
Thanks for your guide.